mutation_reader.cc

```diff
@@ -336,30 +358,59 @@ void mutation_reader_merger::retire_reader(reader_and_last_fragment_kind rk) {
     }
 }
 
-future<> mutation_reader_merger::prepare_next(db::timeout_clock::time_point timeout) {
-    return parallel_for_each(_next, [this, timeout] (reader_and_last_fragment_kind rk) {
-        return (*rk.reader)(timeout).then([this, rk] (mutation_fragment_opt mfo) {
-            if (mfo) {
-                if (mfo->is_partition_start()) {
-                    _reader_heap.emplace_back(rk.reader, std::move(*mfo));
-                    boost::push_heap(_reader_heap, reader_heap_compare(*_schema));
-                } else {
-                    _fragment_heap.emplace_back(rk.reader, std::move(*mfo));
-                    boost::range::push_heap(_fragment_heap, fragment_heap_compare(*_schema));
-                }
-            } else if (_fwd_sm == streamed_mutation::forwarding::yes && rk.last_kind != mutation_fragment::kind::partition_end) {
-                // When in streamed_mutation::forwarding mode we need
-                // to keep track of readers that returned
-                // end-of-stream to know what readers to ff. We can't
-                // just ff all readers as we might drop fragments from
-                // partitions we haven't even read yet.
-                // Readers whoose last emitted fragment was a partition
-                // end are out of data for good for the current range.
-                _halted_readers.push_back(rk);
+bool mutation_reader_merger::in_gallop_mode() const {
+    return _gallop_mode_hits >= gallop_mode_entering_threshold;
+}
+
+future<mutation_reader_merger::needs_merge> mutation_reader_merger::prepare_one(db::timeout_clock::time_point timeout,
+        reader_and_last_fragment_kind rk, handle_galloping handle_galloping) {
```

Comment:
reader_galloping might be a better name.

Reply:
Done, changed